plugin-server/src/worker/ingestion/event-pipeline/4-prepareEventStep.ts

```diff
@@ -1,27 +1,31 @@
 import { PluginEvent } from '@posthog/plugin-scaffold'
 
-import { normalizeEvent } from '../../../utils/event'
+import { IngestionEvent, IngestionPersonData } from '../../../types'
 import { parseEventTimestamp } from '../timestamps'
 import { EventPipelineRunner, StepResult } from './runner'
 
-export async function prepareEventStep(runner: EventPipelineRunner, event: PluginEvent): Promise<StepResult> {
-    // :TRICKY: plugins might have modified the event, so re-sanitize
-    const { ip, site_url, team_id, uuid } = normalizeEvent(event)
+export async function prepareEventStep(
+    runner: EventPipelineRunner,
+    event: PluginEvent,
+    person: IngestionPersonData | undefined
+): Promise<StepResult> {
+    const { ip, site_url, team_id, uuid } = event
     const preIngestionEvent = await runner.hub.eventsProcessor.processEvent(
         String(event.distinct_id),
         ip,
         event,
         team_id,
         parseEventTimestamp(event, runner.hub.statsd),
-        uuid! // it will throw if it's undefined,
+        uuid!, // it will throw if it's undefined,
+        person
     )
 
     await runner.hub.siteUrlManager.updateIngestionSiteUrl(site_url)
 
     if (preIngestionEvent && preIngestionEvent.event !== '$snapshot') {
-        return runner.nextStep('emitToBufferStep', preIngestionEvent)
+        return runner.nextStep('createEventStep', preIngestionEvent)
     } else if (preIngestionEvent && preIngestionEvent.event === '$snapshot') {
-        return runner.nextStep('runAsyncHandlersStep', preIngestionEvent)
+        return runner.nextStep('runAsyncHandlersStep', preIngestionEvent as IngestionEvent)
```

Comment:
`preIngestionEvent as IngestionEvent` is a bit confusing - a comment for now might be sufficient

Reply:
Not sure what to comment on this - ingestion already happened, so we can treat this as an IngestionEvent?